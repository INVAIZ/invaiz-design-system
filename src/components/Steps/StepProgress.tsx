import styled from "@themes/styled";

export interface StepProgressProps {
  currentStep: number;
  totalSteps: number;
  stepWidth?: number;
  stepHeight?: number;
}

interface StepBoxProps {
  isActive: boolean;
  width: number;
  height: number;
}

export default function StepProgress({
  currentStep,
  totalSteps,
  stepWidth = 32,
  stepHeight = 8,
}: StepProgressProps) {
  return (
    <StepProgressContainer>
      {Array.from({ length: totalSteps }, (_, index) => index).map(step => (
        <StepBox
          key={step}
          isActive={step === currentStep}
          width={stepWidth}
          height={stepHeight}
        />
      ))}
    </StepProgressContainer>
  );
}

const StepBox = styled.div<StepBoxProps>`
  width: ${({ width }) => width}px;
  height: ${({ height }) => height}px;
  background-color: ${({ isActive, theme }) =>
    isActive ? theme.color.primary.blue500 : theme.color.grayScale.coolGray200};
  border-radius: 5px;
  margin: 0 8px;
  display: flex;
  justify-content: center;
  align-items: center;
  transition: background 0.3s ease-in-out;
`;

const StepProgressContainer = styled.div`
  display: flex;
  justify-content: center;
  margin-bottom: 2rem;
`;
