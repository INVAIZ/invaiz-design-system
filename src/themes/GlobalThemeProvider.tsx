import type { PropsWithChildren } from "react";
import type { EmotionTheme } from "@themes/emotion-theme";
import type { ColorTheme } from "@themes/colors/color";
// types

import { ThemeProvider } from "@emotion/react";
// React modules

import font from "@themes/font";
import { colorMap, normal, inverse } from "@themes/colors/color";
import { boxShadowMap } from "@themes/box-shadows/box-shadow";
import border from "@themes/border";
import whiteSpace from "@themes/whiteSpace";

interface GlobalThemeProviderProps {
  colorTheme?: ColorTheme;
}

export default function GlobalThemeProvider({
  children,
  colorTheme = "normal",
}: PropsWithChildren<GlobalThemeProviderProps>) {
  const theme: EmotionTheme = {
    normal,
    inverse,
    color: colorMap[colorTheme],
    currentThemeName: colorTheme,

    font,

    boxShadow: boxShadowMap[colorTheme],
    border,
    whiteSpace,
  };
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
}
