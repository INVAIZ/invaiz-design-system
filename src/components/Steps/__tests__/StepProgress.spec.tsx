import { render } from "@tests/test-utils";
import StepProgress from "@components/Steps/StepProgress";
import GlobalThemeProvider from "@themes/GlobalThemeProvider";

const renderWidthTheme = (ui: React.ReactNode) => {
  return render(<GlobalThemeProvider>{ui}</GlobalThemeProvider>);
};

const mockTheme = {
  color: {
    primary: { blue500: "#0066ff" },
    grayScale: { coolGray200: "#ccd7eb" },
  },
};

describe("StepProgress", () => {
  it("정확한 개수의 단계를 렌더링해야 합니다", () => {
    const { container } = renderWidthTheme(
      <StepProgress currentStep={0} totalSteps={5} />,
    );
    const stepProgressContainer = container.firstChild as Element;

    expect(stepProgressContainer.childNodes).toHaveLength(5);
  });

  it("현재 단계를 강조해야 합니다", () => {
    const { container } = renderWidthTheme(
      <StepProgress currentStep={3} totalSteps={6} />,
    );

    const steps = container.firstChild?.childNodes as NodeListOf<ChildNode>;

    const activeStep = steps[3];
    const inactiveStep = steps[0];

    expect(activeStep).toHaveStyle({
      backgroundColor: mockTheme.color.primary.blue500,
    });
    expect(inactiveStep).toHaveStyle({
      backgroundColor: mockTheme.color.grayScale.coolGray200,
    });
  });
});

test("StepProgress Snapshot", () => {
  const currentStep = render(
    <GlobalThemeProvider>
      <StepProgress currentStep={0} totalSteps={5} />
    </GlobalThemeProvider>,
  );
  expect(currentStep).toMatchSnapshot();

  const color = render(
    <GlobalThemeProvider>
      <StepProgress currentStep={3} totalSteps={6} />
    </GlobalThemeProvider>,
  );
  expect(color).toMatchSnapshot();
});
