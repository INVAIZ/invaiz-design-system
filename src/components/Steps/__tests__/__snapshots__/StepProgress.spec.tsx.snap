// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StepProgress Snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-irr4ie"
      >
        <div
          class="css-1v8xmg1"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-irr4ie"
    >
      <div
        class="css-1v8xmg1"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`StepProgress Snapshot 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-irr4ie"
      >
        <div
          class="css-1v8xmg1"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
      </div>
    </div>
    <div>
      <div
        class="css-irr4ie"
      >
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-1v8xmg1"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
        <div
          class="css-w9poon"
          height="8"
          width="32"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-irr4ie"
    >
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-1v8xmg1"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
      <div
        class="css-w9poon"
        height="8"
        width="32"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
